### PR TITLE
Milestone 5: Friendships Version 1

### PR DESCRIPTION
**Pull request by AndresFMoya**
Good day TSEs.

Friendship model was created with associations and all requested features.

Unit and Integration Test were performed with RSpec.

Best!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -8,6 +8,40 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments
   has_many :likes
+  has_many :friendships
```
Comment (mosaaleb) [inline app/models/user.rb]: - [x] Let's add dependent destroy to both `friendships` and `inverse_friendships` associations.

File: app/models/user.rb [hunk 2]
```diff
@@ -8,6 +8,40 @@ class User < ApplicationRecord
...
+    friendships.save
+  end
+
+  def friend?(user)
+    friends.include?(user)
+  end
+
+  def newsfeed
+    friend_ids = 'SELECT friend_id FROM friendships WHERE user_id = :user_id'
+    inverse_friend_ids = 'SELECT user_id FROM friendships WHERE friend_id = :user_id'
```
Comment (mosaaleb) [inline app/models/user.rb]: - [x] Posts written by a user who requested my friendship, and not yet accepted shouldn't appear to my newsfeed page. We need to only include confirmed friendships for `inverse_friend_ids`

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Well done @KerronKing @AndresFMoya :clap:

#### Most requirements are handled, yet there are some needs to be met before moving on.

- [x] sender send a friend request to receiver, sender shouldn't be able to send another request to receiver, receiver shouldn't also be able to send a request to sender. This ensures that if for any reason why we have bugs in the views, no duplicate records will enter our database and sabotage our data integrity.
We can achieve this by adding interchangeable validation to check if the [user_id, friend_id] or [friend_id, user_id] exist in the friendship table. [please take a look at this example](https://stackoverflow.com/questions/29123620/rails-validate-unique-friendship-relationships).


- [x] **`[OPTIONAL]`** Previous step adds model validations to friendships uniqueness. I would suggest to add interchangeable DB validations. You need to know first why DB validations are important [why](https://thoughtbot.com/blog/the-perils-of-uniqueness-validations). 
My previous code partner wrote an article on how to achieve [this](https://www.freecodecamp.org/news/how-to-set-unique-interchangeable-index-constraint-in-rails/).

- [x] We have some failed tests -- Let's fix them.

Screenshot from 2019-11-29 01-06-08

#### Text me on slack if you have any questions.

#### Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Comment by AndresFMoya** [comment]
Good day @mosaaleb. Thanks for your feedback. All requirements were met. Looking to hear your comments. New code review was asked.

Best!

**Review by mosaaleb — APPROVED** [review]
## Well done @AndresFMoya, @KerronKing 👏

#### All requirements are handled. Also, the UI is replica :sparkles:. 

#### You can merge :+1:

#### Good luck with the next milestones.

**Pull request merged**